execution_engine/src/runtime/mod.rs

```diff
@@ -1304,17 +1290,36 @@ where
 
         access_rights.extend(&extended_access_rights);
 
-        if self.is_mint(context_key) {
-            return self.call_host_mint(entry_point.name(), &context_args, access_rights, stack);
-        } else if self.is_handle_payment(context_key) {
-            return self.call_host_handle_payment(
-                entry_point.name(),
-                &context_args,
-                access_rights,
-                stack,
-            );
-        } else if self.is_auction(context_key) {
-            return self.call_host_auction(entry_point.name(), &context_args, access_rights, stack);
+        if let ContractPackageKind::System(system_contract_type) =
+            contract_package.get_package_kind()
+        {
+            match system_contract_type {
+                SystemContractType::Mint => {
+                    return self.call_host_mint(
+                        entry_point.name(),
+                        &context_args,
+                        access_rights,
+                        stack,
+                    );
+                }
+                SystemContractType::HandlePayment => {
+                    return self.call_host_handle_payment(
+                        entry_point.name(),
+                        &context_args,
+                        access_rights,
+                        stack,
+                    );
+                }
+                SystemContractType::StandardPayment => {}
```

Comment:
`self.call_host_standard_payment` is missing here?

Reply:
No, in the previous code tree we only checked if it was the other three system contracts, thus I've left this code path as is

Reply:
standard payment isn't actually a system contract; it's payment logic and is not callable.